mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/domain/JobClusterDefinitionImpl.java

```diff
@@ -75,14 +76,27 @@ public JobClusterDefinitionImpl(@JsonProperty("name") String name,
         this.user = user;
 
         // Todo move the resource cluster label to a property
-        Preconditions.checkNotNull(labels, "labels cannot be empty.");
-        Preconditions.checkArgument(
-            labels.stream()
-                .anyMatch(l ->
-                    l.getName().equalsIgnoreCase(SystemLabels.MANTIS_RESOURCE_CLUSTER_NAME_LABEL.label)),
-            "Missing required label: " + SystemLabels.MANTIS_RESOURCE_CLUSTER_NAME_LABEL.label);
+        if (!isDisabled) {
+            Preconditions.checkNotNull(labels, "labels cannot be empty.");
+            Preconditions.checkArgument(
+                labels.stream()
+                    .anyMatch(l ->
+                        l.getName().equalsIgnoreCase(SystemLabels.MANTIS_RESOURCE_CLUSTER_NAME_LABEL.label)),
+                "Missing required label: " + SystemLabels.MANTIS_RESOURCE_CLUSTER_NAME_LABEL.label);
+        }
     }
 
+    public JobClusterDefinitionImpl(String name,
```

Comment:
nit: @VisibleForTesting? Also, does this need to public or is package-private good enough?

Reply:
Actually it looks like this is only called from the builder. so, maybe private is good enough.

Reply:
it's also used in other path where the create request etc. doesn't have include data about isDisabled thus I gave it a default to maintain the current behavior.